src/Mvc/Mvc.Core/src/Filters/ResponseCacheFilterExecutor.cs

```diff
@@ -106,24 +106,23 @@ public void Execute(FilterContext context)
         else
         {
             string? cacheControlValue;
-            switch (Location)
+
+            if (Location == ResponseCacheLocation.None)
```

Comment:
Remove 'max-age' in 'Cache-Control' Response Header when Location is None ('no-cache').